### PR TITLE
Fixes #721 [dbsnapshot] Allow Azure and other cloud-based server sources with port number.

### LINKED ISSUE #721: dbsnapshot fails with no error code/results
re: ( https://github.com/codesmithtools/Templates/tree/master/Templates/Database/DbSnapshot ) 

Running against an existing Azure MSSQL database that I run other templates against dbsnapshot simply fails.  Also, ran against petshop sample to see if it was an Azure issue, however it fails the same way.   

Environment is windows 10 running in parallels 

image

image

### PR DESCRIPTION
**Pull request by cyberjetx**
The issue wasn't permissions exactly. It was the inability to create a directory from the server name/ server source when it's name is like tcp:myCloudData.database.windows.net,5588.  So I have added a rudimentary check in the ParseServerName function to look for a colon.  

With the example above I choose **myCloudData** as the parsed server name.  I also check for a blank name and if it is I select the computer name as a last resort otherwise we would create a hidden folder eg .MyDatabaseName 

I hope this is clear and I am open to input and suggestion.

### CODE REVIEW

File: Templates/Database/DbSnapshot/DbSnapshot.cst [hunk 1]
```diff
@@ -244,11 +248,24 @@ public string ParseServerName( SchemaExplorer.DatabaseSchema db )
...
+        ) {
+            const RegexOptions ro = RegexOptions.IgnoreCase; // | RegexOptions.IgnorePatternWhitespace | RegexOptions.Multiline | RegexOptions.Compiled ;
+            String regexptrn = @"^.*\:(\w+)(\..*$)";
+            String InPutString = _serverName.Trim();
+            _serverName = Regex.Replace(InPutString, regexptrn, "$1", ro); 
+            CleanName( _serverName ); //insure the name can be used to create a folder safely
+        }
 		
+        if ( _serverName.Trim() == "") { _serverName = CleanName(System.Windows.Forms.SystemInformation.ComputerName); } // edge case 
+
```
Comment (niemyjski) [inline Templates/Database/DbSnapshot/DbSnapshot.cst:268]: I cleaned up the trimming of the server name and noticed clean name on the second part wasn't being applied. Can you please try my suggestions.

```suggestion

    string name = _serverName.Trim();
    if (name.StartsWith(".") || name.StartsWith("(local)") || name.StartsWith("127.0.0.1")) {
        _serverName = CleanName(System.Windows.Forms.SystemInformation.ComputerName);
    } else if (name.Contains(":")) {
        // Allow a cloud source with port  TCP:xxx,1111 or a UDP:xxxx,1433 eg tcp:myCloudData.database.windows.net,5588
        _serverName = CleanName(Regex.Replace(name, @"^.*\:(\w+)(\..*$)", "$1", RegexOptions.IgnoreCase)); 
    }

    if (String.IsNullOrEmpty(_serverName.Trim())) {
        _serverName = CleanName(System.Windows.Forms.SystemInformation.ComputerName); 
    }

```

**Review by niemyjski — CHANGES_REQUESTED** [review]

**Comment by niemyjski** [comment]
Thanks for your PR, I've left some comments above.

**Comment by niemyjski** [comment]
Fixes https://github.com/codesmithtools/Templates/issues/721

**Comment by cyberjetx** [comment]
Changes understood and committed.

**Review by niemyjski — APPROVED** [review]

**Comment by niemyjski** [comment]
Thanks for your PR!

**Pull request merged**